pkg/operator/ceph/client/controller.go

```diff
@@ -150,6 +150,11 @@ func (r *ReconcileCephClient) reconcile(request reconcile.Request) (reconcile.Re
 	if cephClient.Status == nil {
 		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionProgressing)
 	}
+	// update ObservedGeneration in status
+	cephClient.Status.ObservedGeneration = cephClient.ObjectMeta.Generation
+	if err := reporting.UpdateStatus(r.client, cephClient); err != nil {
```

Comment:
@leseb @parth-gr The goal of the observedGeneration is to update it at the end of a reconcile so we know that everything is updated according to that generation, correct? Currently the approach implemented is setting it at the beginning of the reconcile. Maybe this is expected, I just want to be clear on the intention.

To be really clear, we could have two values such as `inProgressGeneration` and `lastSuccessfulGeneration`, but perhaps that's overkill for the purpose.

All calls to `updateStatus()` also need to preserve the current observedGeneration if they aren't setting it. Currently it doesn't seem to be preserved.

Reply:
@travisn updated the changes, I hope now it set the way we discussed.